lib/server.js

```diff
@@ -559,20 +559,33 @@ class WebThingServer {
    * @param {Object} things Things managed by this server -- should be of type
    *                        SingleThing or MultipleThings
    * @param {Number} port Port to listen on (defaults to 80)
+   * @param {String} hostname Optional host name, i.e. mything.com
    * @param {Object} sslOptions SSL options to pass to the express server
    */
-  constructor(things, port, sslOptions) {
+  constructor(things, port, hostname, sslOptions) {
     this.things = things;
     this.name = things.getName();
     this.port = Number(port) || (sslOptions ? 443 : 80);
+    this.hostname = hostname;
+    this.hosts = [
+      '127.0.0.1',
```

Comment:
I am pretty sure that the host header is required to include the port. Why have portless options?

Reply:
@hobinjk Nope. From MDN:

>The Host request header specifies the domain name of the server (for virtual hosting), and (optionally) the TCP port number on which the server is listening.
>
>If no port is given, the default port for the service requested (e.g., "80" for an HTTP URL) is implied.

Reply:
Oh, good to know! Why would I expect a standard to force anything to be in an actual concrete format?